src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -154,15 +175,34 @@ public DashboardWebApplication()
         _app.MapGrpcService<OtlpMetricsService>();
         _app.MapGrpcService<OtlpTraceService>();
         _app.MapGrpcService<OtlpLogsService>();
+
+        static Func<IPEndPoint> CreateEndPointAccessor(ListenOptions options)
+        {
+            // We want to provide a way for someone to get the IP address of an endpoint.
+            // However, if a dynamic port is used, the port is not known until the server is started.
+            // Instead of returning the ListenOption's endpoint directly, we provide a func that returns the endpoint.
+            // The endpoint on ListenOptions is updated after binding, so accessing it via the func after the server
+            // has started returns the resolved port.
+            return () => options.IPEndPoint!;
+        }
     }
 
     public void Run() => _app.Run();
 
+    public Task StartAsync(CancellationToken cancellationToken = default) => _app.StartAsync(cancellationToken);
+
+    public Task StopAsync(CancellationToken cancellationToken = default) => _app.StopAsync(cancellationToken);
+
+    public ValueTask DisposeAsync()
+    {
+        return _app.DisposeAsync();
+    }
+
     private static void ConfigureListenAddresses(KestrelServerOptions kestrelOptions, Uri[] uris, Action<ListenOptions>? configureListenOptions = null)
     {
         foreach (var uri in uris)
         {
-            if (uri.IsLoopback)
+            if (string.Equals(uri.Host, "localhost", StringComparison.OrdinalIgnoreCase))
```

Comment:
Should the `if` be checking for port zero then?

Reply:
I made this change because `ListenLocalHost` doesn't seem like the right thing to do when binding an IPv4 or IPv6 address. If you want both then you'd have an address with `localhost` in it.

Reply:
Thanks, that helps. Would be good to document that in the code. Something to the effect of:

> ListenLocalHost binds both 127.0.0.1 (IPv4) and ::1 (IPv6). We only bind both when the host is "localhost", rather than for all loopback addresses. If only a single protocol version is required, the URI will reflect that.